### PR TITLE
Refactor GrafanaAlertmanager to not depend on Grafana

### PR DESCRIPTION
**Pull request by gotjosh**
This is the first substantial PR at decoupling the Alertmanager logic from the Grafana-specific logic - you can reference that by the removal of the Grafana-based imports from the code and the scope of an interface for that specific logic.

There's a LOT more work TODO (hence the number of comments in the code), but these should be followed up immediately and are very obvious that things don't work. For the ones that we think would be more long term and I'm happy to create an issue to track them.

### CODE REVIEW

File: alerting/grafana_alertmanager.go [hunk 1]
```diff
@@ -135,7 +112,32 @@ type MaintenanceOptions interface {
...
+type Template = template.Template
+type InhibitRule = config.InhibitRule
+type MuteTimeInterval = config.MuteTimeInterval
+type Route = config.Route
+type Integration = notify.Integration
+type DispatcherLimits = dispatch.Limits
+type Notifier = notify.Notifier
+
+// Configuration is an interface for accessing Alertmanager configuration.
+type Configuration interface {
```
Comment (gotjosh) [inline alerting/grafana_alertmanager.go:135]: This is the biggest piece of this work - we've abstracted everything that could be Grafana-specific in an interface, there's probably more than will either go to Grafana or come back here but I've tried to do the minimum for now to get something done.

File: alerting/grafana_alertmanager.go [hunk 2]
```diff
@@ -135,7 +112,32 @@ type MaintenanceOptions interface {
 	MaintenanceFunc() (int64, error)
 }
 
+type Template = template.Template
```
Comment (gotjosh) [inline alerting/grafana_alertmanager.go:125]: https://yourbasic.org/golang/type-alias/

File: alerting/alerts.go [hunk 3]
```diff
@@ -21,11 +22,17 @@ var (
 	ErrGetAlertGroupsBadPayload = fmt.Errorf("unable to retrieve alerts groups")
 )
 
-func (am *GrafanaAlertmanager) GetAlerts(active, silenced, inhibited bool, filter []string, receivers string) (apimodels.GettableAlerts, error) {
+type GettableAlerts = amv2.GettableAlerts
```
Comment (gotjosh) [inline alerting/alerts.go:25]: There will be a better place to organise this, but for now, I want to make the diffs minimal until we know for sure that this works for Grafana. Then we can refactor away at our liking.

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM, let's go!

**Review by JacobsonMT — APPROVED** [review]

**Pull request merged**